gw_eccentricity/eccDefinition.py

```diff
@@ -251,24 +297,178 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
+                {"omegalm_zeroecc": dataDict["omegalm_zeroecc"]})
+        # Look for zeroecc phaselm is amp_phase_omega_dict and compute
+        # omegalm_zeroecc from it if phaselm_zeroecc is present
+        elif "phaselm_zeroecc" in amp_phase_omega_dict \
+             and "t_zeroecc" in dataDict:
+            amp_phase_omega_dict.update(
+                {"omegalm_zeroecc": self.get_omegalm_from_phaselm(
+                    dataDict["t_zeroecc"],
+                    amp_phase_omega_dict["phaselm_zeroecc"])})
+        else:
+            pass
+        return amp_phase_omega_dict
+
+    def get_amplm_from_dataDict(self, dataDict):
+        """Get amplm dict from dataDict.
+
+        Returns the dictionary of amplitudes of waveform modes.
+        """
+        amplm = {}
+        amplm_zeroecc = {}
+        for suffix, ampDict in zip(["", "_zeroecc"], [amplm, amplm_zeroecc]):
+            if "amplm" + suffix in dataDict:
+                # Add the amplitude dictionary from dataDict to ampDict
+                ampDict.update(dataDict["amplm" + suffix])
+            elif "hlm" + suffix in dataDict:
+                # compute amplitude of each hlm mode and add to ampDict
+                for k in dataDict["hlm" + suffix]:
+                    ampDict.update({k: np.abs(dataDict["hlm" + suffix][k])})
+            else:
+                if suffix == "_zeroecc":
+                    # zeroecc amplitude is required only for Residual methods.
+                    pass
+                else:
+                    raise Exception("dataDict should contain at least one of"
+                                    " `'amplm'` or `hlm` dict.")
+        amplmDict = {"amplm": amplm}
+        if amplm_zeroecc:
+            amplmDict.update({"amplm_zeroecc": amplm_zeroecc})
+        return amplmDict
+
+    def get_phaselm_from_dataDict(self, dataDict):
+        """Get phaselm dict from dataDict.
+
+        When the dataDict contains only the complex hlm modes, the phaselm is
+        obtained using the relation hlm = amplm * exp(-i phaselm).
+        """
+        phaselm = {}
+        phaselm_zeroecc = {}
+        for suffix, phaseDict in zip(["", "_zeroecc"],
+                                     [phaselm, phaselm_zeroecc]):
+            if "phaselm" + suffix in dataDict:
+                # Add the phase dictionary from dataDict to phaseDict
+                phaseDict.update(dataDict["phaselm" + suffix])
+            elif "hlm" + suffix in dataDict:
+                # Compute phase of each mode in hlm and add to phaselm
+                for k in dataDict["hlm" + suffix]:
+                    phaseDict.update(
+                        {k: - np.unwrap(
+                            np.angle(dataDict["hlm" + suffix][k]))})
+            elif "omegalm" + suffix in dataDict:
+                # Compute phase of each mode from omega and add to phaseDict
+                for k in dataDict["omegalm" + suffix]:
+                    phaseDict.update(
```

Comment:
This I'm not sure about.
It might be better to require:
Provide either:
1. hlm
2. Alm, philm
3. Alm, philm, omegalm
Instead of doing the integration under the hood, it's probably best to leave it to the user. Most waveforms have easy access to the phase, so having omega and not phase is unusual. My worry is that users that might have the phase might forget about it and just pass omega. And cumtrapz is only second-order accurate, so could eventually become a problem. The reason for allowing omega is the user may have a better omega than we get using a time derivative, in which case, we might as well use that.

Reply:
Changed the code. It now requires the user to provide either hlm or amp and phase. If for example, amp and only omega are provided it will raise exception.